### PR TITLE
Add support for `AVIF` image format

### LINKED ISSUE #17076: Enable `avif` image format
`sharp` supports `avif` natively nowadays, which means that our `format` query parameter and subsequent options in the app support it as well. We should go over all image transformation format uses and make sure `avif` is added to checks and lists.

### PR DESCRIPTION
**Pull request by knulpi**
## Description
This PR adds the `AVIF` image format as possible format to be uploaded and transformed to.
It
* adds `AVIF` as option for transformation presets 
* adds support for images to be transformed to `AVIF` (`AssetsService`)
* adds support for uploading `AVIF` images (`FileService`)
* adds translations (simply `avif`->"AVIF" for all languages)
* refactors supported images into `constants.ts`

Fixes #17076

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [ ] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

**Comment by rijkvanzanten** [comment]
Thanks @knulpi! Mind running Prettier to resolve the linter warnings? Thanks!

**Review by rijkvanzanten — APPROVED** [review]
LGTM!

**Review by paescuj — CHANGES_REQUESTED** [review]
Hey @knulpi, thank you so much for your contribution ❤️ 

Besides some small things, your PR looks very good 👍 

However, I'm sorry to have to tell you that it doesn't work at the moment due to the following issue: https://github.com/libvips/libvips/issues/3274

When uploading or transforming an `avif` file, we receive the following error:
```console
ERROR: source: bad seek to 83048
heif: Invalid input: Unspecified: Bitstream not supported by this decoder (2.0)

    err: {
      "type": "Error",
      "message": "source: bad seek to 83048\nheif: Invalid input: Unspecified: Bitstream not supported by this decoder (2.0)\n",
      "stack":
          Error: source: bad seek to 83048
          heif: Invalid input: Unspecified: Bitstream not supported by this decoder (2.0)

    }
```

I'm afraid we can't merge this pull request until this problem has been solved upstream (and in the shipped `libvips` version of `sharp`).

---

A little tip for your next pull request (which I'm looking forward to 😃): You don't need to update all language files, only the source file (`en-US.yaml`). The other ones are then updated via [Crowdin](https://docs.directus.io/contributing/translations.html#crowdin).

**Comment by knulpi** [comment]
At first, thanks for your feedback!

> When uploading or transforming an `avif` file, we receive the following error:

May I ask what you tested that with and how you proceeded? What I can see is, that _some kinds_ of AVIF (maybe based on the compression used) do raise this error, while others don't. I am trying to find out what property it is exactly so we can investigate further

**Comment by paescuj** [comment]
> At first, thanks for your feedback!

You're welcome!

> > When uploading or transforming an `avif` file, we receive the following error:
> 
> May I ask what you tested that with and how you proceeded?

I've tested it with this sample file: https://github.com/link-u/avif-sample-images/blob/master/fox.profile0.10bpc.yuv420.avif
```console
❯ file fox.profile0.10bpc.yuv420.avif
fox.profile0.10bpc.yuv420.avif: ISO Media, AVIF Image
```

> What I can see is, that _some kinds_ of AVIF (maybe based on the compression used) do raise this error, while others don't. I am trying to find out what property it is exactly so we can investigate further

Hmm that's too bad 😞 Yeah, I guess it somehow depends on the structure / compression although this surprises me a little, since I expected that `avif` files always use AV1...

Nevertheless, I'm afraid that we can't support `avif` like this at the moment (even when some files do work) as this will lead to confusion among users.

**Comment by knulpi** [comment]
> Nevertheless, I'm afraid that we can't support `avif` like this at the moment (even when some files do work) as this will lead to confusion among users.

I agree. Let's hope the issue gets fixed soon.

During debugging I found a potential bug, which caused the API server to crash when reading such an unsupported AVIF image:
https://github.com/directus/directus/blob/dc135f776d3ba453e7764fc5471585bcba0ddc18/api/src/services/files.ts#L142

Shouldn't this be
```ts
if (err) {
  reject(err);
  return;
}
```
to stop execution of the function in case of an error?

**Comment by paescuj** [comment]
> I agree. Let's hope the issue gets fixed soon.

Yep, I hope so too 👍 

> During debugging I found a potential bug, which caused the API server to crash when reading such an unsupported AVIF image:

Wow, good catch! Would you mind fixing this in a pull request? 😃

**Comment by rijkvanzanten** [comment]
@paescuj bummer about the `libvips` issue. Is that a common installation? Do we have a sense of how many environments are affected? Seeing we ship Directus in a Docker image, we could easily get around that by making sure the environment is prepared correctly (that's kinda the point of docker anyways), but wanna get a rough sense of how many direct Node uses would be hindered 🤔

File: api/src/services/files.ts [hunk 1]
```diff
@@ -24,6 +24,7 @@ import { ItemsService } from './items';
 
 // @ts-ignore
 import formatTitle from '@directus/format-title';
+import { SUPPORTED_IMAGE_UPLOAD_FORMATS } from '../constants';
```
Comment (paescuj) [inline api/src/services/files.ts:27]:
```suggestion
import { SUPPORTED_IMAGE_METADATA_FORMATS } from '../constants';
```

**Comment by knulpi** [comment]
@paescuj I've seen your conversation in the mentioned issue. Have you tried building `sharp-libvps` with high bitdepth support as suggested and observed if that solves the errors we are getting? I am currently trying to build it that way and see if that changes anything...

**Comment by paescuj** [comment]
> @paescuj I've seen your conversation in the mentioned issue. Have you tried building `sharp-libvps` with high bitdepth support as suggested and observed if that solves the errors we are getting? I am currently trying to build it that way and see if that changes anything...

Hey @knulpi
Yes, I did that and it's working 👍 I've got a pull request for this in the pipeline, going to push that later on this day.

**Comment by knulpi** [comment]
> Yes, I did that and it's working 👍 I've got a pull request for this in the pipeline, going to push that later on this day.

Nice! I can also confirm now that it works.

**Comment by paescuj** [comment]
> @paescuj bummer about the `libvips` issue. Is that a common installation? Do we have a sense of how many environments are affected? Seeing we ship Directus in a Docker image, we could easily get around that by making sure the environment is prepared correctly (that's kinda the point of docker anyways), but wanna get a rough sense of how many direct Node uses would be hindered 🤔

@rijkvanzanten So it turns out that all environments are affected, as long as the shipped libvips version of sharp is used. This version is intentionally built with limited bit depth support in order to keep the size low, see https://github.com/libvips/libvips/issues/3274#issuecomment-1406341947. That means it will probably not change in the future either.
For this reason I'm in favor of your idea to ship the docker image with our own libvips version 👍
For those users using the installation without docker, I think a corresponding note somewhere in the documentation should be enough.

**Comment by rijkvanzanten** [comment]
Ah must curious. Then a follow up question: how often is the bit depth > 8? Is that a common use case when using AVIFs?

**Comment by paescuj** [comment]
Hard to tell, I'd say AVIF files with bit depth > 8 are much less common, but still a possibility to be reckoned with.
It's worth mentioning that the first sample file I got hold of was a 10bit file.
I also think that the support of higher bit depths in AVIF is one of the strengths over other formats and could become more relevant in the future.

If we enable AVIF with limited bit depth support, I feel that confusion could quickly arise among users:
- Purely from the naming or even with the command `file`, you won't see the depth of the file.
  You'd have to use a tool like `exiftool` to get it.
  So I think many users are not aware of it.
- If you upload or try to transform from such a file in Directus you'd get some non-meaningful error messages:
  - Browser: `500 Internal Server Error`
  - Server:
    ```
    ERROR: source: bad seek to 83048
    heif: Invalid input: Unspecified: Bitstream not supported by this decoder (2.0)
    ```
  Since we cannot parse such files for this very reason, it's also difficult to catch this error reliably.

We could, of course, simply include an appropriate note in the documentation. But I can imagine that sooner or later we'd need our custom build of libvips anyway (e.g. for heic)...

**Comment by knulpi** [comment]
Is there anything I can do to support this?

**Comment by paescuj** [comment]
> Is there anything I can do to support this?

I don't think so right now, we still have to decide how to continue here...

**Comment by rijkvanzanten** [comment]
I'm having a lot of difficulty trying to figure out if 8-bit depth is _enough_ support for us to ship it 🤔 I just don't have the data points required to know if 10+ -bit AVIFs are common enough to a point where people will run into that limitation very often. It might as well be that everything is just kinda 8 by default with only specialized use cases touching the higher bit-rates, I just don't know :| I'm inclined to say there's probably a reason why `sharp` decided that 8-bit is _good enough_ for GA, which we could mirror, as long as we document it somewhere properly. @knulpi what are your thoughts?

**Comment by knulpi** [comment]
> @knulpi what are your thoughts?

I also cannot assess the usage amount of 8+ bits AVIFs. I'd say, however, 8-bit AVIF support is better than no AVIF support. And if we handle the 8+ bits error gracefully, I do not see a reason _not_ to implement this now and not have to fiddle around with the sharp binaries. And if YAGNI gets us and someone has a huge requirement for it, it may be time to reconsider extending the support. That were my thoughts :D

**Comment by rijkvanzanten** [comment]
From what I can tell after a bit of Googling; it looks like the main use case for 8 vs 10 bit is HDR and wider color gamut images (like display-p3) beyond that. This page 
has some good examples https://gregbenzphotography.com/hdr/.

**Comment by rijkvanzanten** [comment]
@knulpi I fully agree with that assessment! Lets do it 🚀 Mind [resolving the CLA](https://github.com/directus/directus/pull/17303#issuecomment-1504112399)? I'll make sure this PR is up to date with the latest changes from main and do a last quick review 👍🏻

**Review by rijkvanzanten — APPROVED** [review]

**Comment by paescuj** [comment]
As just discussed with @rijkvanzanten on Discord I also agree that we should give this a shot and merge it like this 🚀 

> And if we handle the 8+ bits error gracefully [...]

That's however not something we can do as we are dependent on sharp here...

**Review by paescuj — APPROVED** [review]
Thanks @knulpi!

**Pull request merged**